balboa-http/src/main/scala/com/socrata/balboa/server/rest/EntitiesRest.scala

```diff
@@ -1,39 +1,43 @@
 package com.socrata.balboa.server.rest
 
 import java.util.regex.Pattern
-import javax.servlet.http.HttpServletRequest
 
-import com.rojoma.json.ast.JString
 import com.socrata.balboa.metrics.data.DataStoreFactory
-import com.socrata.http.server.implicits._
-import com.socrata.http.server.responses._
-import com.socrata.http.server.{HttpResponse, Service}
+import com.socrata.balboa.server.{Error, ResponseWithType}
+import com.socrata.balboa.server.ResponseWithType.json
 import org.codehaus.jackson.map.annotate.JsonSerialize
 import org.codehaus.jackson.map.{ObjectMapper, SerializationConfig}
+import org.eclipse.jetty.http.HttpStatus.BAD_REQUEST_400
+import org.scalatra.{BadRequest, Ok, Params}
 
 import scala.collection.JavaConverters._
+import scala.util.Try
 
-object EntitiesRest extends Service[HttpServletRequest, HttpResponse] {
+object EntitiesRest {
   val ds = DataStoreFactory.get
 
-  def apply(req: HttpServletRequest): HttpResponse = {
-    val qs = new QueryExtractor(req)
+  def apply(params: Params): ResponseWithType = {
     val predicate: (String => Boolean) =
-      qs[String]("filter").map(_.right.get).map(Pattern.compile).map { pattern => (str: String) => pattern.matcher(str).matches }.getOrElse(_ => true)
+      params.get("filter").map(Pattern.compile)
+        .map { pattern => (str: String) => pattern.matcher(str).matches }.getOrElse(_ => true)
 
     val it = ds.entities().asScala.filter(predicate)
     // backwards-compatibility: -1 == no limit
-    val limit = qs[Int]("limit", -1) match {
-      case Right(l) => l
-      case Left(err) =>
-        return BadRequest ~> ContentType("application/json") ~> Content("""{"error": 400, "message": "Unable to parse limit : """ + JString(err).toString.drop(1).dropRight(1) + """"}""")
-    }
+    val limitString = params.getOrElse("limit", "-1")
+    val limit = Try(limitString.toInt).recover({
+      case err: NumberFormatException =>
+        return ResponseWithType(json,
+          BadRequest(Error(
+            BAD_REQUEST_400,
+            "Unable to parse limit : " + err.toString)))
```

Comment:
Not err.toString. That's implementation dependent detail. And in theory the output of toString can change at any time. (It never does though.) Just a message that indicates the value can't be parsed as an int is sufficient.